my only question was about future plans in https://github.com/kubernetes/kubernetes/pull/83520/files#r342871606, lgtm otherwise

unit test failures do look relevant:

> --- FAIL: TestWatch (1.14s)
>     cacher_test.go:370: (called from line 448)
>     cacher_test.go:371: Expected (ERROR): &v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"", Reason:"Gone", Details:(*v1.StatusDetails)(nil), Code:410}, got: &v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"too old resource version: 1 (2)", Reason:"Expired", Details:(*v1.StatusDetails)(nil), Code:410}
> 
> --- FAIL: TestEmptyWatchEventCache (6.14s)
>     cacher_test.go:370: (called from line 672)
>     cacher_test.go:371: Expected (ERROR): &v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"", Reason:"Gone", Details:(*v1.StatusDetails)(nil), Code:410}, got: &v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"too old resource version: 6 (7)", Reason:"Expired", Details:(*v1.StatusDetails)(nil), Code:410}

/hold
so this doesn't merge on a retest until test failure cause is determined

--- a/staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go
@@ -444,8 +444,8 @@ func TestWatch(t *testing.T) {
 	}
 	defer tooOldWatcher.Stop()
 	// Ensure we get a "Gone" error
-	expectedGoneError := errors.NewGone("").ErrStatus
-	verifyWatchEvent(t, tooOldWatcher, watch.Error, &expectedGoneError)
+	expectedResourceExpiredError := errors.NewResourceExpired("").ErrStatus
+	verifyWatchEvent(t, tooOldWatcher, watch.Error, &expectedResourceExpiredError)
 
 	initialWatcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", fooCreated.ResourceVersion, storage.Everything)
 	if err != nil {
@@ -668,8 +668,8 @@ func TestEmptyWatchEventCache(t *testing.T) {
 			t.Fatalf("Unexpected error: %v", err)
 		}
 		defer watcher.Stop()
-		expectedGoneError := errors.NewGone("").ErrStatus
-		verifyWatchEvent(t, watcher, watch.Error, &expectedGoneError)
+		expectedResourceExpiredError := errors.NewResourceExpired("").ErrStatus
+		verifyWatchEvent(t, watcher, watch.Error, &expectedResourceExpiredError)
 	}
 
 	{

